default/scripting/species/common/focus.macros

```diff
@@ -10,15 +10,37 @@ HAS_RESEARCH_FOCUS
 '''        Focus
             name = "FOCUS_RESEARCH"
             description = "FOCUS_RESEARCH_DESC"
-            location = Planet
+            location = And [
+                Planet
+                Or [
+                    Not EmpireHasAdoptedPolicy empire = Source.Owner name = "PLC_RACIAL_PURITY"
+                    Number low = 1 condition = And [
+                        Planet
+                        OwnedBy empire = Source.Owner
+                        Species name = Source.Species
+                        Contains Building name = "BLD_IMPERIAL_PALACE"
+                    ]
+                ]
+            ]
             graphic = "icons/focus/research.png"
 '''
 
 HAS_INFLUENCE_FOCUS
 '''        Focus
             name = "FOCUS_INFLUENCE"
             description = "FOCUS_INFLUENCE_DESC"
-            location = Planet
+            location = And [
```

Comment:
what actually happens if a focus is dynamically removed? does it stick? does it silently fall back to default focus?

Reply:
I don't know exacly. For the natives I had in the map, they where researching and after adopting policy they got industry focus.

Reply:
I dont quite get the last sentence natives? AI? were researching and switching to industry? influence? 
I meant what happens if after setting influence focus the location condition changes to false (e.g. because they loose their capital). I would guess that the focus stays, but one cant switch back to it. But it is a guess and should probably be tested

Reply:
I have a save where I'm testing the results of this PR.
In that save, my empire is Eaxaw and there is a natives' planet I conquered.
Those natives are set to Research Focus.
When I adopt Racial Purity, those natives that are research focused become industry focused.

In other words, this part apparently works as intended. I don't think I need any help with this. Thank you.

Do you need any other clarification?